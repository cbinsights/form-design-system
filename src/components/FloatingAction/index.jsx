import React from 'react';
import PropTypes from 'prop-types';
import IconButton, { THEMES } from 'components/IconButton';

export const VALID_FAB_THEMES = THEMES.filter((t) => t !== 'ghost');

const FloatingAction = ({
  theme = 'aqua',
  Icon,
  label,
  isActive,
  isLoading,
  ...rest
}) => (
  <div className={`shape--circle elevation--2 display--inlineBlock bgColor--${theme}`}>
    <IconButton
      isActive={isActive}
      isLoading={isLoading}
      Icon={Icon}
      label={label}
      size="m"
      radius="circle"
      theme={theme}
      {...rest}
    />
  </div>
);

FloatingAction.propTypes = {
  /** Used to render a FDS Icon (should only be used for FDS Icons) */
  Icon: PropTypes.func.isRequired,

  /** Accessibility label */
  label: PropTypes.string.isRequired,

  /** Show loading spinner if true */
  isLoading: PropTypes.bool,

  /** Controls active style UI of button */
  isActive: PropTypes.bool,

  /** Theme of IconButton in the FloatingAction */
  theme: PropTypes.oneOf(VALID_FAB_THEMES),
};

export default FloatingAction;
