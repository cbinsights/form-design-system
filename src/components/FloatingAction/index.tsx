import React from 'react';
import Icon from 'components/Icon';
import IconButton from 'components/IconButton';

type VALID_FAB_THEMES = 'aqua' | 'outlined';

export interface FloatingActionProps {
  /** Used to render a FDS Icon (should obnly be used for FDS Icons) */
  Icon: Icon;

  /** Accessibility label */
  label: string;

  /** Show loading spinner if true */
  isLoading?: boolean;

  /** Controls active style UI of button */
  isActive?: boolean;

  /** Theme of IconButton in the FloatingAction VALID_FAB_THEMES */
  theme?: VALID_FAB_THEMES;
}

const FloatingAction = ({
  theme = 'aqua',
  Icon,
  label,
  isActive,
  isLoading,
  ...rest
}: FloatingActionProps): JSX.Element => (
  <div className={`shape--circle elevation--2 display--inlineBlock bgColor--${theme}`}>
    <IconButton
      isActive={isActive}
      isLoading={isLoading}
      Icon={Icon}
      label={label}
      size="m"
      radius="circle"
      theme={theme}
      {...rest}
    />
  </div>
);

export default FloatingAction;
