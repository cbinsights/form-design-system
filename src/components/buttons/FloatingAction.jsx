import React from 'react';
import PropTypes from 'prop-types';
import IconButton, { THEMES } from './IconButton';

export const VALID_FAB_THEMES = THEMES.filter((t) => t !== 'ghost');

const FloatingAction = ({ Icon, label, isActive, isLoading, ...otherProps }) => (
  <div className="shape--circle elevation--3 display--inlineBlock">
    <IconButton
      isActive={isActive}
      isLoading={isLoading}
      Icon={Icon}
      label={label}
      size="m"
      radius="circle"
      {...otherProps}
    />
  </div>
);

FloatingAction.defaultProps = {
  theme: 'aqua',
};

FloatingAction.propTypes = {
  /** Used to render a FDS Icon (should only be used for FDS Icons) */
  Icon: PropTypes.func.isRequired,

  /** Accessibility label */
  label: PropTypes.string.isRequired,

  /** Show loading spinner if true */
  isLoading: PropTypes.bool,

  /** Controls active style UI of button */
  isActive: PropTypes.bool,

  /** Theme of IconButton in the FloatingAction */
  theme: PropTypes.oneOf(VALID_FAB_THEMES),
};

export default FloatingAction;
