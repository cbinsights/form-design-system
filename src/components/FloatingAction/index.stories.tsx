import ApproveIcon from 'icons/react/ApproveIcon';
import { StoryObj } from '@storybook/react';
import FloatingAction, { FloatingActionProps } from '.';

export const Primary: StoryObj<FloatingActionProps> = {
  args: {
    Icon: ApproveIcon,
    label: 'Add Column',
  },
};

export default {
  component: FloatingAction,
  title: 'Components/FloatingAction',
  parameters: {
    docs: {
      fds: {
        attributes: ['rest'],
      },
    },
  },
};
