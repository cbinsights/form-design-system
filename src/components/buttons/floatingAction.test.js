import React from 'react';
import { shallow } from 'enzyme';

import FloatingAction from './FloatingAction';

describe('FloatingAction component', () => {
  it('matches snapshot (set all props)', () => {
    const component = shallow(
      <FloatingAction
        Icon={() => {}}
        label="delete the internet"
      />
    );
    expect(component).toMatchSnapshot();
  });
});
