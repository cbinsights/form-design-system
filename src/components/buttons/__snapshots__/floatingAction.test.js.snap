// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FloatingAction component matches snapshot (set all props) 1`] = `
<div
  className="shape--circle elevation--3 display--inlineBlock"
>
  <IconButton
    Icon={[Function]}
    label="delete the internet"
    radius="circle"
    size="m"
    theme="aqua"
  />
</div>
`;
